src/Symfony/Component/AutoMapper/AutoMapper.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Symfony\Component\AutoMapper;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\AutoMapper\Exception\RuntimeException;
+use Symfony\Component\AutoMapper\Attributes\MapIf;
+use Symfony\Component\AutoMapper\Attributes\MapTo;
+use Symfony\Component\AutoMapper\Attributes\MapWith;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * Object to object mapper.
+ *
+ * @experimental
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class AutoMapper implements AutoMapperInterface
+{
+    public function __construct(private ?PropertyAccessorInterface $propertyAccessor = null, private ?ContainerInterface $serviceLocator = null)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function map(object $object, object|string|null $to = null): mixed
+    {
+        $refl = new \ReflectionClass($object);
+
+        if (!$to) {
+            $to = $this->getAttribute($refl, MapTo::class, true)->to;
+        }
+
+        $arguments = [];
+        if (is_object($to)) {
+            $toRefl = new \ReflectionClass(get_class($to));
+            $mapped = $to;
+        } else {
+            $toRefl = new \ReflectionClass($to);
+            $mapped = $toRefl->newInstanceWithoutConstructor();
+        }
+
+        $constructor = $toRefl->getConstructor();
+        foreach ($constructor?->getParameters() ?? [] as $parameter) {
+            $arguments[$parameter->getName()] = $parameter->isDefaultValueAvailable() ? $parameter->getDefaultValue() : null;
+        }
+
+        foreach ($refl->getProperties() as $property) {
+            $propertyName = $property->getName();
+            $mapTo = $this->getAttribute($property, MapTo::class)?->to ?? $propertyName;
+            if (!$toRefl->hasProperty($mapTo)) {
```

Comment:
So the user DOES need to at least have a `MapTo()` on an arg to activate mapping. That's what you meant by:

> The rest is not "auto".

Reply:
AFAIU, the copy will be done if `$from` and `$to` have props with the same name, even if they have no `MapTo` attributes.

This test just skips properties having no corresponding props in the receiving object.